zerver/tests/test_email_notifications.py

```diff
@@ -1829,6 +1916,20 @@ def test_long_term_idle_user_missed_message(self) -> None:
                 [{"message_id": personal_message_id, "trigger": "private_message"}],
             )
 
+        # Topic Wildcard mention should NOT soft reactivate the user
```

Comment:
I think actually it makes sense to trigger soft-reactivating topic participants -- the rationale being that generally mentioning a topic is going to reach a small set of users who are probably already active, and anyone who is not already active has a decent change of being reactivated by the mention.

Reply:
Makes sense.

I added a separate commit for this. (Last commit)